### PR TITLE
FEATURE: New health check

### PR DESCRIPTION
**Pull request by mhsdesign**
see https://github.com/neos/neos-development-collection/issues/4243 for overview

Thanks to everyone who helped ;)

this is the flow part of it 

neos part: https://github.com/neos/neos-setup/pull/12

how to try it out?

use this branch `feature/new-health-check` for `neos/setup` and `neos/neos-setup`

requires https://github.com/neos/flow-development-collection/pull/3045

```
    "neos/setup": "dev-feature/new-health-check as dev-master",
    "neos/neos-setup": "dev-feature/new-health-check",
```

in case youre trying to use the new setup in a dev collection, it will fail:
> They both replace neos/cli-setup and thus cannot coexist.

because `neos/cli-setup` is not yet removed.

run
`./flow setup`

and navigate to 

`/setup` in your browser ;)

ideally try installing a fresh neos with the help of those tools.

Todo:
- [x] https://github.com/neos/setup/issues/60
- [x] cache invalidation bug: the db settings cache are not emptied so updating the db connection doesnt let the check pass
- [x] test on windows

### CODE REVIEW

**Comment by mhsdesign** [comment]
Todo:

- welcome rename
- welcome is currently only compiletime -> sub request with runtime?

File: Classes/Infrastructure/Healthcheck/BasicRequirementsHealthcheck.php [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    }
+
+    public function getTitle(): string
+    {
+        return 'Basic system requirements';
+    }
+
+    public function execute(HealthcheckEnvironment $environment): Health
+    {
+        $this->environment = $environment;
```
Comment (mhsdesign) [inline Classes/Infrastructure/Healthcheck/BasicRequirementsHealthcheck.php:41]: i dont like it, but whats the alternative? @kitsunet 

and a `setEnviroment` method on the interface (or better like in other places with is_callable) is also not the holy grail ...

Comment (kitsunet) [reply Classes/Infrastructure/Healthcheck/BasicRequirementsHealthcheck.php:41]: Yeah, wouldn't do. Either we introduce options you can set per check and set those options differnetly per environment, or we just disable all checks in production (again via settings) and people decide what they want to activate.

**Comment by mhsdesign** [comment]
Why we run the first check pre-compile time:

- No errors will be thrown when there are invalid php sources (as we dont attempt to require many classes / might be because we trigger the reflection service?)
- On windows the compile boot might not succeed if the php version is not set correclly

**Pull request merged**